### PR TITLE
Add support for custom mongodb commands

### PR DESCRIPTION
**Pull request by SebastianLng**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Extends the MongoDB monitor to allow running custom MongoDB commands and optionally comparing the value against an expected value.

Besides being able to check database content, this is useful to further check the health of MongoDB, especially clusters. For example, the [hello](https://www.mongodb.com/docs/manual/reference/command/hello/) command offers various health information which can be queried and checked with this change.

From a UI standpoint, this adds a command input, a json query input and an expected value input. All are basically the same as comparable HTTP(s) Json query inputs and other DB monitors inputs.
All new fields are optional, and the default command is still the ping command that was used before. All commands are checked for the ok result, so the behaviour without using the new inputs is the same as before.

If the Json Query is set, it'll be applied to the query result. If json query result is undefined, the monitor will produce a failure. The Json query without expected value is basically checking if the value exists. If the expected value is set, it will be compared to the result.

## Type of change

- User interface (UI)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots

image

## Testing
Test MongoDB
`docker run --name test-mongodb -p 27017:27017 -e MONGO_INITDB_ROOT_USERNAME=admin -e MONGO_INITDB_ROOT_PASSWORD=admin mongo`

Connection string: `mongodb://admin:admin@localhost:27017`
Test query: `{ "hello": 1 }`

Here are the tests I ran:
[mongodb-monitor-tests.json](https://github.com/louislam/uptime-kuma/files/14138176/mongodb-monitor-tests.json)

### CODE REVIEW

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Foremost, I think this sort of conflicts with #3919
@chakflying I think this is your area ⇒ I would rather not interfere:
what do you think about this?
Can this be integrated into your plan, or what level of refactoring need to be done to accommodate it?

Given the level of change (basically a rewrite) that this PR adds to the `mongodb` monitor, I think some additional changes are necessary.

- Please extract this monitor to the newer `monitor-type` variant of implementing monitors. see #4395 for an example
- To be make verifying that this works simpler, please attach how you tested it to the description. (`docker run...`, query executed) ^^

**Comment by chakflying** [comment]
We discussed this a bit in #3857. Personally, I think the code change here is relatively small, and is good preparation if we wanted to support this generically in the future. I'm not against merging this if people wanted this feature.

File: server/monitor-types/mongodb.js
Comment (CommanderStorm) [inline server/monitor-types/mongodb.js:15]: nit: This way this is a bit closer

```suggestion
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Overall, I am pretty happy now.
I still need to verify that previously existing monitors work as expected, but looking at the code that should work.

I have some UX remarks about the frontend and about error messages.
Sorry to have split this review into two parts.

File: src/pages/EditMonitor.vue
Comment (SebastianLng) [reply src/pages/EditMonitor.vue:441]: Good idea, especially since a mongodb command is required here, not an typical selection query. Added.

Comment (SebastianLng) [reply src/pages/EditMonitor.vue:441]: To keep the same behaviour as the original mongodb monitor, where just a ping is executed, I made everything optional and made sure every case works. Maybe not optimal but I think the success and error messages are clear enough so people get how it behaves. It also means you can just send a command without requiring to know a specific value that comes back, which makes it simpler for people who still just want to ping, so I like that the inputs are optional in that regard.
An alternative would be to have a separate monitor. Similar to "HTTP(s)" and "HTTP(s) - Json Query", where everything is required in the json query one, but I think that's not really needed here.

File: src/pages/EditMonitor.vue
Comment (SebastianLng) [reply src/pages/EditMonitor.vue:448]: see comment above

File: server/monitor-types/mongodb.js
Comment (SebastianLng) [reply server/monitor-types/mongodb.js:22]: I don't think it's needed. There's no official error field https://www.mongodb.com/docs/manual/reference/method/db.runCommand/#response and 
all errors I tried to produce throw an exception with a nice message, so the user knows what went wrong.

File: server/monitor-types/mongodb.js
Comment (SebastianLng) [reply server/monitor-types/mongodb.js]: This case could also mean that there was not jsonata expression and no expected value, just that the command was successful. I added better success messages to every case instead.

**Review by chakflying — APPROVED** [review]
Tested seems to work okay.

File: src/lang/en.json
Comment (CommanderStorm) [inline src/lang/en.json:888]:
```suggestion
    "mongodbCommandDescription": "Run a MongoDB command against the database. For information about the available commands check out the {documentation}",
    "documentationOf": "{0} Documentation"
```

File: src/pages/EditMonitor.vue
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:443]:
```suggestion
                                        <template #documentation>
                                            <a href="https://www.mongodb.com/docs/manual/reference/command/">{{ $t('documentationOf', ['MongoDB']) }}</a>
                                        </template>
```

**Review by CommanderStorm — APPROVED** [review]
From my side there are no roadblocks.
Works as intended.

**Comment by CommanderStorm** [comment]
Yes, we plan to merge this sometime in v2.x.
I don't know if this can be better/automatically tested via https://github.com/louislam/uptime-kuma/pull/4451
=> would not merge this until that has been investigated

**Comment by CommanderStorm** [comment]
> I don't know if this can be better/automatically tested via #4451

Honestly, such testcases should be build in #4451 first => we can expand this later

Thanks for the new feature!

All changes in this PR work as intended
⇒ merging with junior maintainer approval

**Comment by ekbaramundi** [comment]
Does this also work for MongoDB Atlas?

**Comment by CommanderStorm** [comment]
We have not tested that (we have tested the docker container), but you can report the findings.
I cannot see a reason why this should not work..

**Pull request merged**
